services/submission.js

```diff
@@ -471,35 +471,133 @@ class Submission {
         }
         return this.dataRecordService.listSubmissionNodeTypes(submissionID)
     }
-
+    /**
+     * list Submission Nodes or files
+     * @param {*} params 
+     * @param {*} context 
+     * @returns returnVal object
+     */
     async listSubmissionNodes(params, context) {
         verifySession(context)
-            .verifyInitialized()
-        const result = await this.dataRecordService.submissionNodes(params.submissionID, params.nodeType, 
-            params.first, params.offset, params.orderBy, params.sortDirection);
+            .verifyInitialized();
+
+        //check if submission exists
+        const aSubmission = await findByID(this.submissionCollection, params.submissionID);
+        if(!aSubmission){
+            throw new Error(ERROR.INVALID_SUBMISSION_NOT_FOUND);
+        }
 
         let returnVal = {
-            total: result.total,
+            total: 0,
             properties: [],
             nodes: []
         };
-        if (result.results && result.results.length > 0){
-            let propsSet = new Set();
-            for (let node of result.results) {
-                if (node.parents && node.parents.length > 0) {
-                    for (let parent of node.parents) {
-                        node.props[`${parent.parentType}.${parent.parentIDPropName}`] = parent.parentIDValue;
+        if (params?.nodeType && params?.nodeType !== "Data File") {
+            const result = await this.dataRecordService.submissionNodes(params.submissionID, params.nodeType, 
+                params.first, params.offset, params.orderBy, params.sortDirection);
+    
+            returnVal.total = result.total;
+            if (result.results && result.results.length > 0){
+                let propsSet = new Set();
+                for (let node of result.results) {
+                    if (node.parents && node.parents.length > 0) {
+                        for (let parent of node.parents) {
+                            node.props[`${parent.parentType}.${parent.parentIDPropName}`] = parent.parentIDValue;
+                        }
                     }
+                    if (node.props && Object.keys(node.props).length > 0){
+                        Object.keys(node.props).forEach(propsSet.add, propsSet);
+                    }
+                    node.props = JSON.stringify(node.props);
+                    delete node.parents;
+                    returnVal.nodes.push(node);
                 }
-                if (node.props && Object.keys(node.props).length > 0){
-                    Object.keys(node.props).forEach(propsSet.add, propsSet);
-                }
-                node.props = JSON.stringify(node.props);
-                delete node.parents;
-                returnVal.nodes.push(node);
+                returnVal.properties = Array.from(propsSet);
+            }
+            return returnVal;
+            
+        }
+        else {
```

Comment:
It seems to me the condition of the "else" here is either no "nodeType" or nodeType == "Data File". Is that intention? Should we move condition for nodeType to be inside **if params?.nodeType**?

Reply:
Updated.  Since the nodeType for the query is required so there are only two cases, either "Data File" or not.